### PR TITLE
Property level validation for Management API

### PR DESCRIPTION
**Pull request by kjac**
### Prerequisites

- [x] I have added steps to test this contribution in the description below

### Description

Adds support for property level validation in the management API for document and media editing operations. Any validation errors are returned in an `errors` collection of the resulting problem details, thus mimicking how ASP.NET Core handles request model validation errors.

As content must always be saved despite property validation errors, the API cannot output the validation errors in the actual create and update responses:

- Create must yield `201 Created` and a location header pointing to the newly created resource.
- Update must yield `200 OK` without a response body.

To get around this, a new pair of "validation endpoints" have been introduced for create and update, respectively. These endpoints solely concerns themselves with property level validation.

The PR adds support for outputting an endpoint "operation status". This has been added for all document and media operations.

### Testing this PR

Unstructured notes for test scenarios 😄 

- [ ] JSON path in problem details `errors` for complex editor validation errors
- [ ] Create & update always succeeds despite property editor level validation errors
- [ ] Can validate both create and update models (separate endpoints)
- [ ] Publish performs re-validation and fails with appropriate property level validation errors
- [ ] Validation works for both documents and media
- [ ] Problem details response contains status code for all failure states when working with documents and media - not limited to property level validation

### CODE REVIEW

File: src/Umbraco.Infrastructure/PropertyEditors/BlockEditorValidatorBase.cs [hunk 1]
```diff
@@ -19,35 +20,45 @@ protected IEnumerable<ElementTypeValidationModel> GetBlockEditorDataValidation(B
...
+            new { Path = nameof(BlockValue<TLayout>.SettingsData).ToFirstLowerInvariant(), Items = blockEditorData.BlockValue.SettingsData }
+        };
+
+        foreach (var group in itemDataGroups)
+        {
+            var allElementTypes = _contentTypeService.GetAll(group.Items.Select(x => x.ContentTypeKey).ToArray()).ToDictionary(x => x.Key);
 
-            // now ensure missing properties
-            foreach (IPropertyType elementTypeProp in elementType.CompositionPropertyTypes)
+            for (var i=0; i<group.Items.Count; i++)
```
Comment (Zeegaan) [inline src/Umbraco.Infrastructure/PropertyEditors/BlockEditorValidatorBase.cs:34]:
```suggestion
            for (var i = 0; i < group.Items.Count; i++)
```

**Review by Zeegaan — CHANGES_REQUESTED** [review]
Found a few small nitpicky things 😊

File: src/Umbraco.Cms.Api.Management/Controllers/Media/CreateMediaController.cs [hunk 2]
```diff
@@ -39,25 +39,25 @@ public class CreateMediaController : MediaControllerBase
...
-        Attempt<IMedia?, ContentEditingOperationStatus> result = await _mediaEditingService.CreateAsync(model, CurrentUserKey(_backOfficeSecurityAccessor));
+        MediaCreateModel model = _mediaEditingPresentationFactory.MapCreateModel(requestModel);
+        Attempt<MediaCreateResult, ContentEditingOperationStatus> result = await _mediaEditingService.CreateAsync(model, CurrentUserKey(_backOfficeSecurityAccessor));
 
         return result.Success
-            ? CreatedAtId<ByKeyMediaController>(controller => nameof(controller.ByKey), result.Result!.Key)
+            ? CreatedAtId<ByKeyMediaController>(controller => nameof(controller.ByKey), result.Result.Content!.Key)
             : ContentEditingOperationStatusResult(result.Status);
     }
 }
```
[the remark this answers is not in the document]
Comment (kjac) [reply src/Umbraco.Cms.Api.Management/Controllers/Media/CreateMediaController.cs:63]: You're absolutely right. That's an oversight. Good catch 👍

File: src/Umbraco.Cms.Api.Management/Controllers/Document/UpdateDocumentControllerBase.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            ContentPermissionResource.WithKeys(ActionUpdate.ActionLetter, id, cultures),
+            AuthorizationPolicies.ContentPermissionByResource);
+
+        if (!authorizationResult.Succeeded)
+        {
+            return Forbidden();
+        }
+
+        IContent? content = await _contentEditingService.GetAsync(id);
+        if (content == null)
```
[the remark this answers is not in the document]
Comment (kjac) [reply src/Umbraco.Cms.Api.Management/Controllers/Document/UpdateDocumentControllerBase.cs]: Fixed in latest

File: src/Umbraco.Cms.Api.Management/Controllers/Media/CreateMediaController.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (kjac) [reply src/Umbraco.Cms.Api.Management/Controllers/Media/CreateMediaController.cs]: Fixed in latest

File: src/Umbraco.Cms.Api.Management/Controllers/Media/UpdateMediaControllerBase.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+            MediaPermissionResource.WithKeys(id),
+            AuthorizationPolicies.MediaPermissionByResource);
+
+        if (!authorizationResult.Succeeded)
+        {
+            return Forbidden();
+        }
+
+        IMedia? media = await _mediaEditingService.GetAsync(id);
+        if (media == null)
```
[the remark this answers is not in the document]
Comment (kjac) [reply src/Umbraco.Cms.Api.Management/Controllers/Media/UpdateMediaControllerBase.cs]: Fixed in latest

File: src/Umbraco.Core/PropertyEditors/Validation/JsonPathValidator.cs [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        if (validationResult is not INestedValidationResults nestedValidationResults || nestedValidationResults.ValidationResults.Any() is false)
+        {
+            return;
+        }
+
+        if (validationResult is IJsonPathValidationResult jsonPathValidationResult)
+        {
+            current.Children.Add(new JsonPathValidationTreeItem
+            {
+                JsonPath = $"{current.JsonPath}.{jsonPathValidationResult.JsonPath}"
```
[the remark this answers is not in the document]
Comment (kjac) [reply src/Umbraco.Core/PropertyEditors/Validation/JsonPathValidator.cs]: Fixed in latest

File: src/Umbraco.Infrastructure/PropertyEditors/BlockEditorValidatorBase.cs (hunk 1 above)
Comment (kjac) [reply src/Umbraco.Infrastructure/PropertyEditors/BlockEditorValidatorBase.cs]: Fixed in latest

File: src/Umbraco.Core/Models/ContentEditing/ContentCreateResult.cs [hunk 6]
```diff
@@ -0,0 +1,5 @@
+﻿namespace Umbraco.Cms.Core.Models.ContentEditing;
+
+public class ContentCreateResult : ContentCreateResultBase<IContent>
+{
+}
```
[the remark this answers is not in the document]
Comment (kjac) [reply src/Umbraco.Core/Models/ContentEditing/ContentCreateResult.cs:5]: As agreed - no primary constructors (yet)

**Review by Zeegaan — APPROVED** [review]
LGTM 🚀

**Pull request merged**